src/Commands/GenerateModule.php

```diff
@@ -0,0 +1,321 @@
+    public function hook".$hookfunction."(\$params) {
+        //TODO
+    }
+    ";
+        }
+
+        $stringEntryPoint = "";
+        $stringEntryPoint.= "<?php
+if (!defined('_CAN_LOAD_FILES_')) {
+    exit;
+}
+
+require_once __DIR__ . '/vendor/autoload.php';
+
+use PrestaShop\PrestaShop\Core\Module\WidgetInterface;
+
+/**
+* Class ".$filenameFirstletterCaps."
+*/
+
+class ".$filenameFirstletterCaps." extends Module implements WidgetInterface
+{
+
+    const MODULE_NAME = '".$fileName."';
+
+    public \$templateFile;
+
+    public function __construct()
+    {
+        \$this->name = '".$filenameCamel."';
+        \$this->author = '".$author."';
+        \$this->version = '1.0.0';
+        \$this->need_instance = 0;
+        \$this->bootstrap = true;
+        parent::__construct();
+
+        \$this->displayName = '".$filenameCamel."';
+        \$this->description = '".$description."';
+        \$this->secure_key = Tools::encrypt(\$this->name);
+
+        \$this->ps_versions_compliancy = array('min' => '1.7.5.0', 'max' => _PS_VERSION_);
+        \$this->templateFile = 'module:".$filenameCamel."/views/templates/hook/';
+
+        \$this->tabs = [
+        ];
+    }
+
+    public function install()
+    {
+        return parent::install()".$hookStringInstall."
+            && \$this->installTab();
+    }
+
+    public function uninstall()
+    {
+        return parent::uninstall()
+            && \$this->uninstallTab();
+    }
+
+    public function installTab()
+    {
+        //TODO
+    }
+
+    public function uninstallTab()
+    {
+        //TODO
+    }
+
+    public function enable(\$force_all = false)
+    {
+        return parent::enable(\$force_all)
+            && \$this->installTab();
+    }
+
+    public function disable(\$force_all = false)
+    {
+        return parent::disable(\$force_all)
+            && \$this->uninstallTab();
+    }
+    ".$hookStringFunction."
+    public function renderWidget(\$hookName, array \$configuration)
+    {
+        //TODO
+    }
+
+    public function getWidgetVariables(\$hookName, array \$configuration)
+    {
+        //TODO
+    }
+}";
+        $this->whriteCode($filesystem, $fileName, $stringEntryPoint);
+
+//        $io->text("table 1");
+//        $io->table(
+//            ['Nom', 'Type', 'Null'],
+//            [
+//                ['id_truc', 'int(11)', 'Y'],
+//                ['date_bidule', 'datetime', 'Y'],
+//                ['machin', 'varchar(20)', 'Y'],
+//            ]
+//        );
+    }
+
+    protected function whriteCode($filesystem, $fileName, $string)
+    {
+        $filesystem->dumpFile(_PS_MODULE_DIR_."/".$fileName."/".$fileName.".php", $string);
+    }
+
+    protected function menuHook($io)
+    {
+        $io->listing([
+            '1. Add a hook',
+            '2. Modify a hook',
+            '3. Delete a hook',
+            '4. Display the list of hooks',
+            '5. Leave',
+        ]);
+        $numMenu = $io->ask('Enter a number');
```

Comment:
why not using https://symfony.com/doc/3.4/components/console/helpers/questionhelper.html#let-the-user-choose-from-a-list-of-answers ?

Reply:
done